compiler/accelerators/rocc.py

```diff
@@ -35,52 +37,84 @@ def lower_acc_setup(
         setup_op: acc.SetupOp, acc_op: acc.AcceleratorOp
     ) -> Sequence[Operation]:
         xcustom_acc = 3  # hardcoded to 3 for now
-        vals = create_pairs(setup_op, acc_op.field_names())
+        vals = create_pairs(setup_op)
+        # Only pass on the field names that are set in the current setup
+        instructions = set([name[:-4:] for name, _ in setup_op.iter_params()])
+        current_fields = {
+            key: val for key, val in acc_op.field_items() if key[:-4:] in instructions
+        }.items()
         # Create the sequence of all operations that need to be emitted
-        return combine_pairs_to_ops(acc_op.field_items(), vals, xcustom_acc)
+        return combine_pairs_to_ops(current_fields, vals, xcustom_acc)
 
 
-def combine_pairs_to_ops(field_items, values, xcustom_acc):
+def create_pairs(
+    fields_op: acc.LaunchOp | acc.SetupOp,
+) -> dict[str, tuple[SSAValue, SSAValue]]:
+    """
+    For a given RoCC launch or setup op, return a map that maps a single RoCC
+    instruction to pairs of two SSAValues
+
+    For setup ops, this can retrace back previous setup state if necessary
+    (i.e. if one of the two operands of an instruction gets dedupped)
+    """
+    # Make a set of all the unique instruction names in the current operation
+    instructions = set([name[:-4:] for name, _ in fields_op.iter_params()])
```

Comment:
what does this `name[:-4:]` wizardry do?

Reply:
wait, is this just the same as `name[:-4]`? so just removing the last part (.rsx)? this would make sense, but the second colon confused me 🙃

Reply:
I'm really not a fan of string parsing for logic like this, although I don't see a better solution for this case right now

Reply:
@jorendumoulin the current system is quite brittle indeed :(

> although I don't see a better solution for this case right now
Same!

I've been wondering how I can add a verifier inside the interface to check for some things: i.e.
Making sure there is a rs1 and rs2 behind everything, making sure all of those pairs are pointing to the same address, but I'm not sure where to put those (making the acceleratorop verifier call into the acceleratorinterface seems like a no-no?)